homeassistant/components/matter/adapter.py

```diff
@@ -20,6 +21,14 @@
     from matter_server.client.models.node import MatterEndpoint, MatterNode
 
 
+def get_clean_name(name: str | None) -> str | None:
+    """Strip spaces and null char from the name."""
+    if name is None:
+        return name
```

Comment:
Not sure when `name` can be `None`.

Reply:
it can be

Reply:
It is not `None` if the field in the json is `null`.

Reply:
yes it is

Reply:
https://github.com/home-assistant/core/pull/92073/files#diff-575d4dfe303cf5c235f5252a13f31362e27c1be2977689a060805c6fd4479958R53

If I set the name `null`. It will not pass `None` to `get_clean_name`

Reply:
I corrected the comment above. `null` returns an empty string.

Reply:
that is strange and sounds like an issue with the json decoder ? null in json should translate to None in python.

Reply:
Yeah, strange. I can see it is read as `None`, but somewhere it seems to be converted to an empty string.